src/ucs/datastruct/ptr_array.c

```diff
@@ -172,37 +190,128 @@ unsigned ucs_ptr_array_insert(ucs_ptr_array_t *ptr_array, void *value,
     }
 
     /* Get the first item on the free list */
-    elem_index = ptr_array->freelist;
-    ucs_assert(elem_index != UCS_PTR_ARRAY_SENTINEL);
-    elem = &ptr_array->start[elem_index];
+    element_index = ptr_array->freelist;
+    ucs_assert(element_index != UCS_PTR_ARRAY_SENTINEL);
+
+    elem = &ptr_array->start[element_index];
+
+    /* Remove from free list and populate */
+    ptr_array->freelist =
+        ucs_ptr_array_freelist_get_next_params(*elem, &size_free_ahead);
```

Comment:
you can use `ucs_ptr_array_freelist_get_next` here because `size_free` is not needed

Reply:
Fixed.